src/data/polynomial/derivative.lean

```diff
@@ -301,6 +301,14 @@ begin
   { simp [ih p.derivative, iterate_derivative_neg, derivative_comp, pow_succ], },
 end
 
+lemma eval_prod_derivative {S : multiset R} {r : R} (hr : r ∈ S) :
```

Comment:
This name sounds quite generic, but the lemma is quite specific. This is longer, but maybe not too long?
```suggestion
lemma eval_multiset_prod_X_sub_C_derivative {S : multiset R} {r : R} (hr : r ∈ S) :
```
Do we already have a `multiset_prod_derivative` lemma that shows that the derivative of a product is a sum of derivatives of products over `S.erase _`? That seems like a good way to forward-rewrite in a situation like this. That might turn the proof of this lemma into a `simp only`-oneliner.

Reply:
I didn't find anything about the derivative of a product over a `multiset` (not even over a `finset)`. Let me add it.

Reply:
I've added `derivative_prod`, but it doesn't help in making the proof shorter: it turns the derivative of the product in a sum, and then one has to realize that only one term is different from zero, so I think it becomes longer that it actually is.

I am leaving `derivative_prod` there even if it is not really relevant for this PR (it is in any case a useful lemma to have).